apps/dashboard/src/components/common/SideBar.tsx

```diff
@@ -0,0 +1,149 @@
+import { profileSelectors, resetUserProfile } from "@/redux/user/profile";
+import { Route } from "@/lib/route";
+import { SideBarAccordion } from "@/components/common/SideBarAccordion";
+import { NavAccordionItem, NavOption } from "@/types/navbar";
+import SideBarItem from "@/components/common/SideBarItem";
+
+const framerSidebarBackground = {
+  initial: { x: "30rem", y: "0rem" },
+  animate: { x: "0rem", y: "0rem" },
+  exit: { x: "20rem", y: "0rem" },
+  transition: { duration: 0.2 },
+};
+
+const SidebarNav: React.FC<{
+  options: (NavOption | NavAccordionItem)[];
+}> = ({ options }) => {
+  const dispatch = useAppDispatch();
+  const intl = useFormatMessage();
+  const profile = useAppSelector(profileSelectors.user);
+
+  const button = useRef<HTMLDivElement>(null);
+  const { toggle: toggleTheme, theme } = useTheme();
+  const navigate = useNavigate();
+  const {
+    toggle: toggleMenu,
+    open,
+    ref,
+  } = useClosableRef<HTMLUListElement>(button.current);
+
+  const logout = useCallback(() => {
+    removeToken();
+    dispatch(resetUserProfile());
+    navigate(Route.Login);
+  }, [dispatch]);
+
+  return (
+    <nav className="relative">
+      <div
+        ref={button}
+        className="fixed flex flex-col items-center justify-between h-screen p-2 border-l"
+      >
+        <Button
+          onClick={toggleMenu}
+          size={ButtonSize.Small}
+          type={ButtonType.Text}
+          className="!p-2"
+        >
+          <Sidebar className="w-8 h-8" />
+        </Button>
+        <Button type={ButtonType.Text} onClick={toggleMenu}>
+          {profile?.image ? (
+            <div className="overflow-hidden border-2 border-white rounded-full">
+              <img src={profile?.image} className="w-10 h-10 rounded-full" />
+            </div>
+          ) : (
+            <User className="w-8 h-8" />
+          )}
+        </Button>
+      </div>
+
+      <AnimatePresence mode="wait" initial={false}>
+        {open ? (
+          <div
+            className={cn(
+              "fixed right-0 top-0 h-screen w-screen backdrop-blur-sm z-[1000]"
+            )}
+          >
+            <motion.ul
+              {...framerSidebarBackground}
+              className={cn(
+                "bg-background-dialog h-full w-72 justify-between",
+                "border-l border-border-overlay rounded-l-md shadow-lg",
+                "flex flex-col items-center p-6"
+              )}
+              ref={ref}
+            >
+              <div className="flex items-center w-full gap-2">
+                <div>
+                  {profile?.image ? (
+                    <div className="overflow-hidden border-2 border-white rounded-full">
+                      <img
+                        src={profile.image}
+                        className="w-10 h-10 rounded-full"
+                      />
+                    </div>
+                  ) : (
+                    <User className="w-8 h-8" />
+                  )}
```

Comment:
I can see this pattern repeated twice. You can consider moving it into a shared component and then reuse it.
```suggestion
                  <UserImage />
```